src/project/mod.rs

```diff
@@ -393,10 +399,14 @@ impl Project {
     }
 
     /// Returns the custom location of pypi-name-mapping
-    pub fn pypi_name_mapping_source(&self) -> &'static MappingSource {
-        self.manifest
-            .pypi_name_mapping_source()
-            .expect("mapping source should be ok")
+    pub fn pypi_name_mapping_source(&self) -> MappingSource {
```

Comment:
You could return a `&MappingSource` here, this would eliminate a clone. A user of the function can clone it themselfs if they want to.